scopes/preview/preview/preview.main.runtime.tsx

```diff
@@ -78,9 +78,9 @@ export class PreviewMain {
     return this.workspace?.getTempDir(PreviewAspect.id) || DEFAULT_TEMP_DIR;
   }
 
-  async getPreview(component: Component): Promise<PreviewArtifact> {
+  async getPreview(component: Component): Promise<PreviewArtifact | undefined> {
     const artifacts = await this.builder.getArtifactsVinylByExtension(component, PreviewAspect.id);
-    if (!artifacts.length) throw new PreviewArtifactNotFound(component.id);
+    if (!artifacts.length) return undefined;
```

Comment:
@GiladShoham - I'm returning undef instead of throwing. I checked the usages, and typescript is ok with this.